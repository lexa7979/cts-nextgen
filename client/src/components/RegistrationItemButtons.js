/* eslint-disable react/prop-types */
/* eslint-disable no-use-before-define */

import React from "react";
import * as yup from "yup";

import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";

const propsSchema = yup.object( {
	formikBag: yup.object().required(),
} );

export default RegistrationItemButtons;

/**
 * @param	{object}	props
 *
 * @returns	{object}	React component
 */
function RegistrationItemButtons( props ) {
	propsSchema.strict().validate( props );

	const { dirty, isValid, isSubmitting } = props.formikBag;

	return (
		<Grid item container direction="row" justify="flex-end" spacing={2}>
			<Grid item>
				<Button type="submit" variant="contained" color="primary" disabled={!dirty || !isValid || isSubmitting}>
					Submit
				</Button>
			</Grid>
			<Grid item>
				<Button type="reset" variant="contained" disabled={!dirty}>
					Reset
				</Button>
			</Grid>
		</Grid>
	);
}
