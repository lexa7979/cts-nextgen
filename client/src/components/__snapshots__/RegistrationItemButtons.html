
<!DOCTYPE html>
<html style="height: 100%;">
  <head>
  </head>
  <body style="display: flex; flex-flow: column nowrap; justify-content: center; align-items: center; height: 100%;">
    <h2>
      Component RegistrationItemButtons - when rendering into snapshots with valid properties - delivers expected result  (-> check snapshot)
    </h2>
    <!--[ Content from unit test: ]-->
    <div class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-justify-xs-flex-end">
      <div class="MuiGrid-root MuiGrid-item">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary Mui-disabled Mui-disabled"
                tabindex="-1"
                type="submit"
                disabled
        >
          <span class="MuiButton-label">
            Submit
          </span>
        </button>
      </div>
      <div class="MuiGrid-root MuiGrid-item">
        <button class="MuiButtonBase-root MuiButton-root MuiButton-contained Mui-disabled Mui-disabled"
                tabindex="-1"
                type="reset"
                disabled
        >
          <span class="MuiButton-label">
            Reset
          </span>
        </button>
      </div>
    </div>
    <!--[ End of included content ]-->
    <br>
    <br>
    &lt;RegistrationItemButtons formikBag={exampleFormikBag} /&gt;
  </body>
</html>
